File: src/Java_P/List/ArrayList_To_LinkedList/LinkedList/Doubly_LinkedLists.java
package Java_P.List.ArrayList_To_LinkedList.LinkedList;

import java.util.LinkedList;

public class Doubly_LinkedLists {
    public static void main(String[]args){
        LinkedList<String> list = new LinkedList<String>();
        LinkedList<String> list1 = new LinkedList<String>();

        list.addFirst("This");
        list.addLast("is");
        list.add("a");
        list.add("list");
        System.out.println(list);

        list1.addFirst("Munna");
        list1.add("Raju");
        list1.addLast("Sohan");
        System.out.println(list1);

        //All object is add
        list.addAll(list1);
        System.out.println(list);

        //loops iterator
        for (String i :list){
            System.out.print(i+",");
        }
        System.out.println();

        //removeFirst=First element delete
        list.removeFirst();
        System.out.println("removeFirst:"+list);

        //removeLast=Last element delete
        list.removeLast();
        System.out.println("removeLast:"+list);

        //element duplicate
        list.add("is");
        list.clone();
        System.out.println("element duplicate:"+list);

        //all element clean
        list.clear();
        System.out.println("All Data is Clear"+list);
    }
}

File: src/Kotlin/List/LinkedList/Doubly_LinkedLists.kt
package Kotlin.List.LinkedList

import java.util.LinkedList

fun main() {
    val ls = LinkedList<Int>()
    val ls1 = LinkedList<Int>()

    ls.addFirst(10)
    ls.add(20)
    ls.add(2,40)
    ls.add(3,50)
    ls.add(4,60)
    ls.addLast(70)

    ls1.addFirst(10)
    ls1.add(20)
    ls1.add(2,40)
    ls1.add(3,50)
    ls1.add(4,60)
    ls1.addLast(70)

    println(ls)

    //All object is add
    ls.addAll(ls1)
    println(ls)

    //loops iterator
    for (i in ls){
        print("$i ")
    }
    println()

    //removeFirst=First element delete
    ls.removeFirst()
    println("RemoveFirst: $ls")

    //removeLast=Last element delete
    ls.removeLast()
    println("RemoveLast: $ls")

    //element duplicate
     ls.clone()
    println("Clone: $ls")

    //all element clean
    ls.clear()
    println(ls)
}